doozerlib/distgit.py

```diff
@@ -935,6 +937,17 @@ def _extract_container_build_logs(self, task_id):
         except OSError as e:
             self.logger.warning("Exception while trying to extract build logs in {}: {}".format(logs_dir, e))
 
+    def _add_missing_pkgs(self, pkg_name):
+        """
+        add missing packages to runtime.missing_pkgs set and print it out for
+        `brew tag-build` eg: from `rhaos-4.0-rhel-7-candidate` to `rhaos-4.1-rhel-7-candidate` manually
+        :param pkg_name: Missing packages like: No package (.*) available
+        """
+        latest = self.metadata.exec_latest_build_info(pkg_name)
```

Comment:
This queries the current candidate tag for the RPM name that was missing. There are several reasons why that's never going to return anything:

1. It is the RPM name, and brew only deals in package names. They might be the same, but often they're not.
2. It might not even be a name. It could be a version lock e.g. `python-foo-3.*` or `golang >= 1.10`. It might be a requirement that's actually provided by an RPM of a completely different name.
3. More importantly, this looks for it in the current candidate tag. If it wasn't found during the build, then it's not in the current tag; if anywhere, it's in the past one.

So the missing package should be displayed whether or not you can find anything in brew that provides it, so that a human can search for it. The "try tagging this" advice would be additional, and you'd build that by querying the entire list of latest packages (`brew list-tagged --latest`) from the previous tag  (you can get MAJOR.MINOR from https://github.com/openshift/ocp-build-data/blob/openshift-4.1/group.yml#L178 and decrement MINOR to help determine the previous tag), listing the RPMs that those packages include (`brew buildinfo`), and then searching a dict of those results for the missing RPMs.

I think for the first pass you could just list everything that matched the pattern, and follow up with a PR to actually recommend what to tag to satisfy what's missing.

Reply:
Yes, I'll remove the exec_latest_build in this pr